[glib] `x64_osx` failed in ci

`glib` in triplet `x64_osx` failed to compile, blocking this PR. #28467  
I'm not familiar with this port, please help
  
[failure logs for x64-osx.zip](https://github.com/microsoft/vcpkg/files/10283871/failure.logs.for.x64-osx.zip)

```
[203/577] /Library/Developer/CommandLineTools/usr/bin/cc -Igio/kqueue/libkqueue.a.p -Igio/kqueue -I../src/glib-2-802473e24b.clean/gio/kqueue -I. -I../src/glib-2-802473e24b.clean -Iglib -I../src/glib-2-802473e24b.clean/glib -Igmodule -I../src/glib-2-802473e24b.clean/gmodule -Igio -Igobject -I/Users/vagrant/Data/installed/x64-osx/include -fcolor-diagnostics -Wall -Winvalid-pch -Wextra -Wpedantic -std=gnu99 -O0 -g -D_GNU_SOURCE -fno-strict-aliasing -DG_ENABLE_DEBUG -Wimplicit-fallthrough -Wmisleading-indentation -Wunused -Wno-unused-parameter -Wno-pedantic -Wno-format-zero-length -Wno-variadic-macros -Werror=format=2 -Werror=init-self -Werror=missing-include-dirs -Werror=pointer-arith -Werror=unused-result -Wstrict-prototypes -Wno-bad-function-cast -Werror=implicit-function-declaration -Werror=missing-prototypes -Werror=pointer-sign -Wno-string-plus-int -fPIC -isysroot /Library/Developer/CommandLineTools/SDKs/MacOSX12.1.sdk -g '-DG_LOG_DOMAIN="GLib-GIO"' '-DGIO_LAUNCH_DESKTOP="/Users/vagrant/Data/packages/glib_x64-osx/debug/libexec/gio-launch-desktop"' '-DGIO_MODULE_DIR="/Users/vagrant/Data/packages/glib_x64-osx/debug/lib/gio/modules"' '-DLOCALSTATEDIR="/Users/vagrant/Data/packages/glib_x64-osx/debug/var"' -DGIO_COMPILATION -MD -MQ gio/kqueue/libkqueue.a.p/kqueue-missing.c.o -MF gio/kqueue/libkqueue.a.p/kqueue-missing.c.o.d -o gio/kqueue/libkqueue.a.p/kqueue-missing.c.o -c ../src/glib-2-802473e24b.clean/gio/kqueue/kqueue-missing.c
FAILED: gio/kqueue/libkqueue.a.p/kqueue-missing.c.o 
/Library/Developer/CommandLineTools/usr/bin/cc -Igio/kqueue/libkqueue.a.p -Igio/kqueue -I../src/glib-2-802473e24b.clean/gio/kqueue -I. -I../src/glib-2-802473e24b.clean -Iglib -I../src/glib-2-802473e24b.clean/glib -Igmodule -I../src/glib-2-802473e24b.clean/gmodule -Igio -Igobject -I/Users/vagrant/Data/installed/x64-osx/include -fcolor-diagnostics -Wall -Winvalid-pch -Wextra -Wpedantic -std=gnu99 -O0 -g -D_GNU_SOURCE -fno-strict-aliasing -DG_ENABLE_DEBUG -Wimplicit-fallthrough -Wmisleading-indentation -Wunused -Wno-unused-parameter -Wno-pedantic -Wno-format-zero-length -Wno-variadic-macros -Werror=format=2 -Werror=init-self -Werror=missing-include-dirs -Werror=pointer-arith -Werror=unused-result -Wstrict-prototypes -Wno-bad-function-cast -Werror=implicit-function-declaration -Werror=missing-prototypes -Werror=pointer-sign -Wno-string-plus-int -fPIC -isysroot /Library/Developer/CommandLineTools/SDKs/MacOSX12.1.sdk -g '-DG_LOG_DOMAIN="GLib-GIO"' '-DGIO_LAUNCH_DESKTOP="/Users/vagrant/Data/packages/glib_x64-osx/debug/libexec/gio-launch-desktop"' '-DGIO_MODULE_DIR="/Users/vagrant/Data/packages/glib_x64-osx/debug/lib/gio/modules"' '-DLOCALSTATEDIR="/Users/vagrant/Data/packages/glib_x64-osx/debug/var"' -DGIO_COMPILATION -MD -MQ gio/kqueue/libkqueue.a.p/kqueue-missing.c.o -MF gio/kqueue/libkqueue.a.p/kqueue-missing.c.o.d -o gio/kqueue/libkqueue.a.p/kqueue-missing.c.o -c ../src/glib-2-802473e24b.clean/gio/kqueue/kqueue-missing.c
In file included from ../src/glib-2-802473e24b.clean/gio/kqueue/kqueue-missing.c:26:
In file included from ../src/glib-2-802473e24b.clean/gio/kqueue/kqueue-helper.h:26:
In file included from ../src/glib-2-802473e24b.clean/gio/glocalfilemonitor.h:26:
In file included from ../src/glib-2-802473e24b.clean/gio/gfilemonitor.h:30:
In file included from ../src/glib-2-802473e24b.clean/gio/giotypes.h:30:
../src/glib-2-802473e24b.clean/gio/gioenums.h:31:10: fatal error: 'gio/gio-visibility.h' file not found
#include <gio/gio-visibility.h>
         ^~~~~~~~~~~~~~~~~~~~~~
1 error generated.
[204/577] /Library/Developer/CommandLineTools/usr/bin/cc -Igio/kqueue/libkqueue.a.p -Igio/kqueue -I../src/glib-2-802473e24b.clean/gio/kqueue -I. -I../src/glib-2-802473e24b.clean -Iglib -I../src/glib-2-802473e24b.clean/glib -Igmodule -I../src/glib-2-802473e24b.clean/gmodule -Igio -Igobject -I/Users/vagrant/Data/installed/x64-osx/include -fcolor-diagnostics -Wall -Winvalid-pch -Wextra -Wpedantic -std=gnu99 -O0 -g -D_GNU_SOURCE -fno-strict-aliasing -DG_ENABLE_DEBUG -Wimplicit-fallthrough -Wmisleading-indentation -Wunused -Wno-unused-parameter -Wno-pedantic -Wno-format-zero-length -Wno-variadic-macros -Werror=format=2 -Werror=init-self -Werror=missing-include-dirs -Werror=pointer-arith -Werror=unused-result -Wstrict-prototypes -Wno-bad-function-cast -Werror=implicit-function-declaration -Werror=missing-prototypes -Werror=pointer-sign -Wno-string-plus-int -fPIC -isysroot /Library/Developer/CommandLineTools/SDKs/MacOSX12.1.sdk -g '-DG_LOG_DOMAIN="GLib-GIO"' '-DGIO_LAUNCH_DESKTOP="/Users/vagrant/Data/packages/glib_x64-osx/debug/libexec/gio-launch-desktop"' '-DGIO_MODULE_DIR="/Users/vagrant/Data/packages/glib_x64-osx/debug/lib/gio/modules"' '-DLOCALSTATEDIR="/Users/vagrant/Data/packages/glib_x64-osx/debug/var"' -DGIO_COMPILATION -MD -MQ gio/kqueue/libkqueue.a.p/kqueue-helper.c.o -MF gio/kqueue/libkqueue.a.p/kqueue-helper.c.o.d -o gio/kqueue/libkqueue.a.p/kqueue-helper.c.o -c ../src/glib-2-802473e24b.clean/gio/kqueue/kqueue-helper.c
FAILED: gio/kqueue/libkqueue.a.p/kqueue-helper.c.o 
/Library/Developer/CommandLineTools/usr/bin/cc -Igio/kqueue/libkqueue.a.p -Igio/kqueue -I../src/glib-2-802473e24b.clean/gio/kqueue -I. -I../src/glib-2-802473e24b.clean -Iglib -I../src/glib-2-802473e24b.clean/glib -Igmodule -I../src/glib-2-802473e24b.clean/gmodule -Igio -Igobject -I/Users/vagrant/Data/installed/x64-osx/include -fcolor-diagnostics -Wall -Winvalid-pch -Wextra -Wpedantic -std=gnu99 -O0 -g -D_GNU_SOURCE -fno-strict-aliasing -DG_ENABLE_DEBUG -Wimplicit-fallthrough -Wmisleading-indentation -Wunused -Wno-unused-parameter -Wno-pedantic -Wno-format-zero-length -Wno-variadic-macros -Werror=format=2 -Werror=init-self -Werror=missing-include-dirs -Werror=pointer-arith -Werror=unused-result -Wstrict-prototypes -Wno-bad-function-cast -Werror=implicit-function-declaration -Werror=missing-prototypes -Werror=pointer-sign -Wno-string-plus-int -fPIC -isysroot /Library/Developer/CommandLineTools/SDKs/MacOSX12.1.sdk -g '-DG_LOG_DOMAIN="GLib-GIO"' '-DGIO_LAUNCH_DESKTOP="/Users/vagrant/Data/packages/glib_x64-osx/debug/libexec/gio-launch-desktop"' '-DGIO_MODULE_DIR="/Users/vagrant/Data/packages/glib_x64-osx/debug/lib/gio/modules"' '-DLOCALSTATEDIR="/Users/vagrant/Data/packages/glib_x64-osx/debug/var"' -DGIO_COMPILATION -MD -MQ gio/kqueue/libkqueue.a.p/kqueue-helper.c.o -MF gio/kqueue/libkqueue.a.p/kqueue-helper.c.o.d -o gio/kqueue/libkqueue.a.p/kqueue-helper.c.o -c ../src/glib-2-802473e24b.clean/gio/kqueue/kqueue-helper.c
In file included from ../src/glib-2-802473e24b.clean/gio/kqueue/kqueue-helper.c:29:
In file included from ../src/glib-2-802473e24b.clean/gio/glocalfile.h:26:
In file included from ../src/glib-2-802473e24b.clean/gio/giotypes.h:30:
../src/glib-2-802473e24b.clean/gio/gioenums.h:31:10: fatal error: 'gio/gio-visibility.h' file not found
#include <gio/gio-visibility.h>
         ^~~~~~~~~~~~~~~~~~~~~~
1 error generated.
[205/577] /Library/Developer/CommandLineTools/usr/bin/cc -Iglib/libglib-2.0.a.p -Iglib -I../src/glib-2-802473e24b.clean/glib -I. -I../src/glib-2-802473e24b.clean -I/Users/vagrant/Data/installed/x64-osx/include -I/Users/vagrant/Data/installed/x64-osx/debug/lib/pkgconfig/../../../include -fvisibility=hidden -fcolor-diagnostics -Wall -Winvalid-pch -Wextra -Wpedantic -std=gnu99 -O0 -g -D_GNU_SOURCE -fno-strict-aliasing -DG_ENABLE_DEBUG -Wimplicit-fallthrough -Wmisleading-indentation -Wunused -Wno-unused-parameter -Wno-pedantic -Wno-format-zero-length -Wno-variadic-macros -Werror=format=2 -Werror=init-self -Werror=missing-include-dirs -Werror=pointer-arith -Werror=unused-result -Wstrict-prototypes -Wno-bad-function-cast -Werror=implicit-function-declaration -Werror=missing-prototypes -Werror=pointer-sign -Wno-string-plus-int -fPIC -isysroot /Library/Developer/CommandLineTools/SDKs/MacOSX12.1.sdk -g '-DG_LOG_DOMAIN="GLib"' -DGLIB_COMPILATION -MD -MQ glib/libglib-2.0.a.p/gvarianttype.c.o -MF glib/libglib-2.0.a.p/gvarianttype.c.o.d -o glib/libglib-2.0.a.p/gvarianttype.c.o -c ../src/glib-2-802473e24b.clean/glib/gvarianttype.c
[206/577] /Library/Developer/CommandLineTools/usr/bin/cc -Iglib/libglib-2.0.a.p -Iglib -I../src/glib-2-802473e24b.clean/glib -I. -I../src/glib-2-802473e24b.clean -I/Users/vagrant/Data/installed/x64-osx/include -I/Users/vagrant/Data/installed/x64-osx/debug/lib/pkgconfig/../../../include -fvisibility=hidden -fcolor-diagnostics -Wall -Winvalid-pch -Wextra -Wpedantic -std=gnu99 -O0 -g -D_GNU_SOURCE -fno-strict-aliasing -DG_ENABLE_DEBUG -Wimplicit-fallthrough -Wmisleading-indentation -Wunused -Wno-unused-parameter -Wno-pedantic -Wno-format-zero-length -Wno-variadic-macros -Werror=format=2 -Werror=init-self -Werror=missing-include-dirs -Werror=pointer-arith -Werror=unused-result -Wstrict-prototypes -Wno-bad-function-cast -Werror=implicit-function-declaration -Werror=missing-prototypes -Werror=pointer-sign -Wno-string-plus-int -fPIC -isysroot /Library/Developer/CommandLineTools/SDKs/MacOSX12.1.sdk -g '-DG_LOG_DOMAIN="GLib"' -DGLIB_COMPILATION -MD -MQ glib/libglib-2.0.a.p/gprintf.c.o -MF glib/libglib-2.0.a.p/gprintf.c.o.d -o glib/libglib-2.0.a.p/gprintf.c.o -c ../src/glib-2-802473e24b.clean/glib/gprintf.c
[207/577] /Library/Developer/CommandLineTools/usr/bin/cc -Igio/kqueue/libkqueue.a.p -Igio/kqueue -I../src/glib-2-802473e24b.clean/gio/kqueue -I. -I../src/glib-2-802473e24b.clean -Iglib -I../src/glib-2-802473e24b.clean/glib -Igmodule -I../src/glib-2-802473e24b.clean/gmodule -Igio -Igobject -I/Users/vagrant/Data/installed/x64-osx/include -fcolor-diagnostics -Wall -Winvalid-pch -Wextra -Wpedantic -std=gnu99 -O0 -g -D_GNU_SOURCE -fno-strict-aliasing -DG_ENABLE_DEBUG -Wimplicit-fallthrough -Wmisleading-indentation -Wunused -Wno-unused-parameter -Wno-pedantic -Wno-format-zero-length -Wno-variadic-macros -Werror=format=2 -Werror=init-self -Werror=missing-include-dirs -Werror=pointer-arith -Werror=unused-result -Wstrict-prototypes -Wno-bad-function-cast -Werror=implicit-function-declaration -Werror=missing-prototypes -Werror=pointer-sign -Wno-string-plus-int -fPIC -isysroot /Library/Developer/CommandLineTools/SDKs/MacOSX12.1.sdk -g '-DG_LOG_DOMAIN="GLib-GIO"' '-DGIO_LAUNCH_DESKTOP="/Users/vagrant/Data/packages/glib_x64-osx/debug/libexec/gio-launch-desktop"' '-DGIO_MODULE_DIR="/Users/vagrant/Data/packages/glib_x64-osx/debug/lib/gio/modules"' '-DLOCALSTATEDIR="/Users/vagrant/Data/packages/glib_x64-osx/debug/var"' -DGIO_COMPILATION -MD -MQ gio/kqueue/libkqueue.a.p/gkqueuefilemonitor.c.o -MF gio/kqueue/libkqueue.a.p/gkqueuefilemonitor.c.o.d -o gio/kqueue/libkqueue.a.p/gkqueuefilemonitor.c.o -c ../src/glib-2-802473e24b.clean/gio/kqueue/gkqueuefilemonitor.c
FAILED: gio/kqueue/libkqueue.a.p/gkqueuefilemonitor.c.o 
/Library/Developer/CommandLineTools/usr/bin/cc -Igio/kqueue/libkqueue.a.p -Igio/kqueue -I../src/glib-2-802473e24b.clean/gio/kqueue -I. -I../src/glib-2-802473e24b.clean -Iglib -I../src/glib-2-802473e24b.clean/glib -Igmodule -I../src/glib-2-802473e24b.clean/gmodule -Igio -Igobject -I/Users/vagrant/Data/installed/x64-osx/include -fcolor-diagnostics -Wall -Winvalid-pch -Wextra -Wpedantic -std=gnu99 -O0 -g -D_GNU_SOURCE -fno-strict-aliasing -DG_ENABLE_DEBUG -Wimplicit-fallthrough -Wmisleading-indentation -Wunused -Wno-unused-parameter -Wno-pedantic -Wno-format-zero-length -Wno-variadic-macros -Werror=format=2 -Werror=init-self -Werror=missing-include-dirs -Werror=pointer-arith -Werror=unused-result -Wstrict-prototypes -Wno-bad-function-cast -Werror=implicit-function-declaration -Werror=missing-prototypes -Werror=pointer-sign -Wno-string-plus-int -fPIC -isysroot /Library/Developer/CommandLineTools/SDKs/MacOSX12.1.sdk -g '-DG_LOG_DOMAIN="GLib-GIO"' '-DGIO_LAUNCH_DESKTOP="/Users/vagrant/Data/packages/glib_x64-osx/debug/libexec/gio-launch-desktop"' '-DGIO_MODULE_DIR="/Users/vagrant/Data/packages/glib_x64-osx/debug/lib/gio/modules"' '-DLOCALSTATEDIR="/Users/vagrant/Data/packages/glib_x64-osx/debug/var"' -DGIO_COMPILATION -MD -MQ gio/kqueue/libkqueue.a.p/gkqueuefilemonitor.c.o -MF gio/kqueue/libkqueue.a.p/gkqueuefilemonitor.c.o.d -o gio/kqueue/libkqueue.a.p/gkqueuefilemonitor.c.o -c ../src/glib-2-802473e24b.clean/gio/kqueue/gkqueuefilemonitor.c
In file included from ../src/glib-2-802473e24b.clean/gio/kqueue/gkqueuefilemonitor.c:37:
In file included from ../src/glib-2-802473e24b.clean/gio/gfilemonitor.h:30:
In file included from ../src/glib-2-802473e24b.clean/gio/giotypes.h:30:
../src/glib-2-802473e24b.clean/gio/gioenums.h:31:10: fatal error: 'gio/gio-visibility.h' file not found
#include <gio/gio-visibility.h>
         ^~~~~~~~~~~~~~~~~~~~~~
1 error generated.
```

--- a/ports/glib/portfile.cmake
+++ b/ports/glib/portfile.cmake
@@ -4,7 +4,7 @@ string(REGEX MATCH "^([0-9]*[.][0-9]*)" GLIB_MAJOR_MINOR "${VERSION}")
 vcpkg_download_distfile(GLIB_ARCHIVE
     URLS "https://download.gnome.org/sources/glib/${GLIB_MAJOR_MINOR}/glib-${VERSION}.tar.xz"
     FILENAME "glib-${VERSION}.tar.xz"
-    SHA512 0402c063975680ff2385876f521b37aa4cc599d2570eb79976ad2a1b530e47a086d514fe122fd870b4a8f7358f48c926285694d153cc2c32cf6963ed2d5da9d9
+    SHA512 4338bf3e42ccbf3679f60b917194070040ff94ba3643121e8916180d6949f3b8cc308b6ad73912bebcb53a29920954bcfb2216bacca0503473a897f1fd023981
 )
 
 vcpkg_extract_source_archive(SOURCE_PATH
@@ -28,10 +28,6 @@ if (libmount IN_LIST FEATURES)
     list(APPEND OPTIONS -Dlibmount=enabled)
 else()
     list(APPEND OPTIONS -Dlibmount=disabled)
-endif()
-
-if(VCPKG_TARGET_IS_WINDOWS)
-    list(APPEND OPTIONS -Diconv=external)
 endif()
 
 vcpkg_configure_meson(

--- a/ports/glib/vcpkg.json
+++ b/ports/glib/vcpkg.json
@@ -1,6 +1,6 @@
 {
   "name": "glib",
-  "version": "2.75.0",
+  "version": "2.75.1",
   "description": "Portable, general-purpose utility library.",
   "homepage": "https://developer.gnome.org/glib/",
   "license": "LGPL-2.1-or-later",

--- a/versions/baseline.json
+++ b/versions/baseline.json
@@ -2669,7 +2669,7 @@
       "port-version": 0
     },
     "glib": {
-      "baseline": "2.75.0",
+      "baseline": "2.75.1",
       "port-version": 0
     },
     "glibmm": {

--- a/versions/g-/glib.json
+++ b/versions/g-/glib.json
@@ -1,6 +1,11 @@
 {
   "versions": [
     {
+      "git-tree": "8f7840ed02d20169b1610123b0f35463c81c523d",
+      "version": "2.75.1",
+      "port-version": 0
+    },
+    {
       "git-tree": "d3b428570b820ec1fbedbb37a651c1e1ddcbb91f",
       "version": "2.75.0",
       "port-version": 0

